PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ["hr.work.location"]
-    _inherit = ['res.partner']
-    _inherit = ['res.users']

Total Changes: 9

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_homeworking/models/hr_employee.py
+++ b/addons/hr_homeworking/models/hr_employee.py
@@ -6,7 +6,7 @@ from .hr_homeworking import DAYS
 
 
 class HrEmployeeBase(models.AbstractModel):
-    _inherit = ["hr.employee.base"]
+    _inherit = "hr.employee.base"
 
     monday_location_id = fields.Many2one('hr.work.location', string='Monday')
     tuesday_location_id = fields.Many2one('hr.work.location', string='Tuesday')

--- a/addons/hr_homeworking/models/hr_homeworking.py
+++ b/addons/hr_homeworking/models/hr_homeworking.py
@@ -6,6 +6,7 @@ DAYS = ['monday_location_id', 'tuesday_location_id', 'wednesday_location_id', 't
 
 
 class HrEmployeeLocation(models.Model):
+    _name = 'hr.employee.location'
     _description = "Employee Location"
 
     work_location_id = fields.Many2one('hr.work.location', required=True, string="Location")

--- a/addons/hr_homeworking/models/hr_work_location.py
+++ b/addons/hr_homeworking/models/hr_work_location.py
@@ -7,7 +7,7 @@ from odoo.addons.hr_homeworking.models.hr_homeworking import DAYS
 
 
 class HrWorkLocation(models.Model):
-    _inherit = ["hr.work.location"]
+    _inherit = "hr.work.location"
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_used_by_employee(self):

--- a/addons/hr_homeworking/models/res_partner.py
+++ b/addons/hr_homeworking/models/res_partner.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class ResPartner(models.Model):
-    _inherit = ['res.partner']
+    _inherit = 'res.partner'
 
     def _compute_im_status(self):
         super()._compute_im_status()

--- a/addons/hr_homeworking/models/res_users.py
+++ b/addons/hr_homeworking/models/res_users.py
@@ -5,7 +5,7 @@ from .hr_homeworking import DAYS
 
 
 class ResUsers(models.Model):
-    _inherit = ['res.users']
+    _inherit = 'res.users'
 
     monday_location_id = fields.Many2one("hr.work.location", related="employee_id.monday_location_id", readonly=False, string='Monday')
     tuesday_location_id = fields.Many2one("hr.work.location", related="employee_id.tuesday_location_id", readonly=False, string='Tuesday')
